src/cargo/core/resolver/mod.rs

```diff
@@ -820,27 +850,36 @@ impl RemainingCandidates {
         }
     }
 
+    /// Attempts to find another candidate to check from this list.
+    ///
+    /// This method will attempt to move this iterator forward, returning a
+    /// candidate that's possible to activate. The `cx` argument is the current
+    /// context which determines validity for candidates returned, and the `dep`
+    /// is the dependency listing that we're activating for.
+    ///
+    /// If successful a `(Candidate, bool)` pair will be returned. The
+    /// `Candidate` is the candidate to attempt to activate, and the `bool` is
+    /// an indicator of whether there are remaining candidates to try of if
+    /// we've reached the end of iteration.
+    ///
+    /// If we've reached the end of the iterator here then `Err` will be
+    /// returned. The error will contain a map of package id to conflict reason,
+    /// where each package id was filtered out from the original list for the
```

Comment:
I think this has the structure backwards, where each package id that has already been activated that causes something to be filtered out to the reason it caused a candidate to be skipped.